src/sources/http/__tests__/index.ts

```diff
@@ -0,0 +1,79 @@
+import fs from 'fs';
+import path from 'path';
+import HttpSource, { buildUri } from '..';
+import createMockServer from '../../../test/utils/createMockServer';
+import { Stream } from 'stream';
+import toMatchImageSnapshot from '../../../test/utils/toMatchImageSnapshot';
+
+expect.extend({ toMatchImageSnapshot });
+
+describe('HTTP Source', () => {
+  describe('buildUri', () => {
+    const paths = [
+      'http://spacechop.com/:image',
+      'http://localhost:8080/:image',
+    ];
+
+    const params = {
+      image: 'test.jpg',
+    };
+
+    const expected = [
+      'http://spacechop.com/test.jpg',
+      'http://localhost:8080/test.jpg',
+    ];
+
+    for (let i in paths) {
+      const path = paths[i];
+      it(`should work with path: ${path}`, () => {
+        expect(buildUri(path, params)).toBe(expected[i]);
+      });
+    }
+  });
+
+  describe('config', () => {
+    const config = { root: 'http://localhost:9000/:image' };
+    const instance = new HttpSource(config);
+    const listener = jest.fn();
+    const handler = (req, res) => {
+      const { image } = req.params;
+      const filename = path.join(__dirname, '../../../test/assets', image);
+      const exists = fs.existsSync(filename);
+      listener(req.originalUrl);
+      res.status(exists ? 200 : 404);
+      if (exists && req.method === 'GET') {
+        const stream = fs.createReadStream(filename);
+        stream.pipe(res);
+      } else {
+        res.end();
+      }
+    };
+    let server;
+
+    beforeAll(async() => {
+      server = await createMockServer(9000, handler, '/:image');
+    });
+    afterAll(async () => {
+      await server.close();
+    });
+
+    it('should save config', () => {
+      expect(instance.config).toEqual(
+        expect.objectContaining(config),
+      );
+    });
+
+    it('should resolve exists', async () => {
```

Comment:
Proposed new name for this test: .exists should resolve to true if image exists

Reply:
changing to: should resolve to true if image exists